### PR TITLE
Improve CI build times and stability with a containerized service

### PR DESCRIPTION
**Pull request by phenaproxima**
At the moment, most of our CI job is tied up in installing the monster that is Acquia CMS. I've created a simple container package of it, all it needs is a database import to get up and running; let's use that instead to save time.

### CODE REVIEW

**Review by phenaproxima — COMMENTED** [review]
A little self-review to explain some of this.

File: .github/workflows/ci.yml
Comment (lauriii) [reply .github/workflows/ci.yml:27]: How should we keep this up to date in future? Should we give at least some other folks permissions to push to this for future use?

File: drupal/.env
Comment (lauriii) [reply drupal/.env:5]: We should document how the database dump is created so that if we need to update it in future, we know how to do that.

**Review by lauriii — CHANGES_REQUESTED** [review]

File: drupal/.env
Comment (phenaproxima) [reply drupal/.env:5]: Yup, working on making this repeatable.

File: .github/workflows/ci.yml
Comment (phenaproxima) [reply .github/workflows/ci.yml:27]: Honestly, it should be on Acquia CMS to provide container images of their software. They can use my work here as a model. Once I've finished writing the build scripts, it should be possible to keep it pretty easily up to date. I'm open to granting push access to one or two Acquia CMS maintainers, but ideally they'd just provide their own "official" images.

Comment (lauriii) [reply .github/workflows/ci.yml:27]: I think we must give someone else access there just for the long term maintainability of this package. It's not a massive deal even if we lose access to the image but ideally we would not have to redo that work in future 😅

Comment (phenaproxima) [reply .github/workflows/ci.yml:27]: I don't object, but to be clear, this image is not meant to be long-term maintainable. We need to coordinate with the Acquia CMS team to come up with a plan here. (I don't think that should block merging this PR, but we should alert them of this need.)

File: .github/workflows/ci.yml
Comment (vishalkhode1) [reply .github/workflows/ci.yml:39]: Hi @phenaproxima I would suggest on not to store db dump (or any tests artifacts) in repo (unless it's really important). Though I noticed the size of file is very small. Because in the past, we used to do the same in Acquia CMS, which made the repo size large. And now we are pulling it on run-time i.e from s3 buckets.

Comment (phenaproxima) [reply .github/workflows/ci.yml:39]: It's very small because it's gzipped. When expanded, it's a lot larger, although it's not TOO big because it doesn't contain any cache data.

It's absolutely critical to store the database dump, because the container does not do a full install of Acquia CMS from the ground up -- it's simply too long and involved of a process, and the entire point of containerizing it was to skip over that in favor of just importing it when we need it.

File: .github/workflows/ci.yml
Comment (lauriii) [reply .github/workflows/ci.yml:27]: I have now access to the image so that at least addresses my immediate concerns. I'd like to still see further work to make maintaining the image easier but I think what we have there now works for now.

File: .github/workflows/ci.yml
Comment (lauriii) [reply .github/workflows/ci.yml:39]: Having 466kb DB dump in the repository is not ideal but I think we can figure out a better solution for this in future. I don't think we should block this PR on this given that this PR would improve our CI build times and stability.

**Review by lauriii — APPROVED** [review]

**Pull request merged**
